Gitra/Controller/Setting/View Model/SettingMenuViewModel.swift

```diff
@@ -0,0 +1,88 @@
+//
+//  SettingViewModel.swift
+//  Gitra
+//
+//  Created by Yahya Ayyash on 05/04/22.
+//
+
+import Foundation
+
+class SettingViewModel {
+    var data: SettingMenuViewModel
+    
+    init() {
+        data = SettingMenuViewModel(setting: SettingsDatabase.shared.getSettings())
+    }
+}
+
+extension SettingViewModel {
+    func settingListCount() -> Int {
+        return data.child?.count ?? 0
+    }
+    
+    func settingForRow(at index: Int) -> SettingMenuViewModel {
+        return data.child?[index] ?? SettingMenuViewModel(setting: SettingMenu(title: "Out of range.", type: .none))
+    }
+    
+    func toggleSettings(value: Int, forKey key: SettingKey) {
+        // TODO: Move User Defaults
+        UserDefaults.standard.set(value, forKey: key.rawValue)
+    }
+}
+
+struct SettingMenuViewModel {
```

Comment:
I don't think it's necessary untuk ngebuat struct baru khusus untuk dipakai di viewmodel. Kinda defeats the whole purpose of having the viewmodel as model interpreter. Jadi method di extension ini sepertinya bisa dipindahin langsung ke viewmodel utama, diaksesnya per index. Banyak yang cukup unnecessary juga sepertinya wrapping varnya, jadi tidak perlu sampe dibuat kyk gini. wdyt?